ui-react/src/components/CheckForUpdates.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useState, useEffect, useContext } from 'react';
+import { Button, ProgressBar, Spinner } from 'react-bootstrap';
+import {
+  checkForUpdate, getAppInfo, openExternal, downloadUpdate
+} from '../eventsBridge';
+import { StoreContext } from '../Store';
+import { styles } from './CheckForUpdates.module.css';
+
+const UPDATE_STATES = {
+  INIT: 'INIT',
+  LOADING: 'LOADING',
+  ERROR: 'ERROR',
+  NEW_VERSION_AVAILABLE: 'NEW_VERSION_AVAILABLE',
+  NO_NEW_VERSION: 'NO_NEW_VERSION',
+  READY_TO_INSTALL: 'READY_TO_INSTALL',
+};
+
+type UpdateInfo = {
+  version: string
+}
+
+function CheckForUpdates() {
+
+  const [updateState, setUpdateState] = useState(UPDATE_STATES.INIT);
+  const [updateInfo, setUpdateInfo] = useState<UpdateInfo>();
+  const store = useContext(StoreContext);
+
+  const checkForUpdates = async () => {
+    setUpdateState(UPDATE_STATES.LOADING);
+    try {
+      const info = await checkForUpdate();
+      setUpdateInfo(info);
+      setUpdateState(info ? UPDATE_STATES.NEW_VERSION_AVAILABLE : UPDATE_STATES.NO_NEW_VERSION);
+    } catch (error) {
+      setUpdateState(UPDATE_STATES.ERROR);
```

Comment:
Log the error?
Swallowing errors makes it hard to understand the root cause of issues

Reply:
Do you mean to log it to the sentry?
Just `console.log(error);`  causing an eslint error:
[](https://eslint.org/docs/rules/no-console)

Reply:
I think we should disable this eslint rule.
De facto, we need to be able to see the root cause when there is an issue, and since this is not a fancy product, but a real tool we give to people I see no harm in logging the errors to the console. WDYT?